crates/core/c8y_smartrest/src/operations.rs

```diff
@@ -98,7 +98,8 @@ fn get_operations(dir: impl AsRef<Path>, cloud_name: &str) -> Result<Operations,
     let mut operations = Operations::default();
 
     let path = dir.as_ref().join(&cloud_name);
-    let dir_entries = fs::read_dir(&path)?
+    let dir_entries = fs::read_dir(&path)
+        .map_err(|_| OperationsError::InvalidOperationName(PathBuf::from(&path)))?
```

Comment:
That wild card match for all errors from `read_dir` being mapped to `OperationsError::InvalidOperationName` might lead to some misleading errors when the dir doesn't even exist. The error message of `InvalidOperationName` is "Cannot extract the operation name from the path". That doesn't sound right when the file itself is missing or when the mapper doesn't have the rights to read that file.

Either we need a more generic `OperationsError` enum with a generic message or the mapping should be more specific rather than `|_|`. I understand that `std::io::Error` is not very handy for such specific mappings. In that case, the former might be the way out.